src/glib/mine/signalproc.cpp

```diff
@@ -1241,6 +1241,298 @@ PJsonVal TOnlineHistogram::SaveJson() const {
 	return Result;
 }
 
+void TTDigest::Update(const TFlt& V) {
+	const TFlt Count = 1;
+	Update(V, Count);
+}
+
+void TTDigest::Update(const TFlt& V, const TFlt& Count) {
+	if (TempLast >= TempWeight.Len()) {
+		MergeValues();
+	}
+	TInt N_ = TempLast++;
+	TempWeight[N_] = Count;
+	TempMean[N_] = V;
+	UnmergedSum += Count;
+	MergeValues();
+}
+TInt TTDigest::GetClusters() const {
+	return Mean.Len();
+}
+TFlt TTDigest::GetQuantile(const TFlt& Q) const {
+		//MergeValues();
+
+		TFlt Total = TotalSum;
+		TInt N_ = Last;
+		TFltV U = Mean;
+	    TFltV W = Weight;
+	    TFltV C = MergeMean;
+	    TInt I;
```

Comment:
- we don't use forward type declariations, rather use int I = Bisect...
- try to come up with variable names that have meaning and possible add some comments on what is being done here
- why do we need to copy Weight and MergeMean? The function is const...